### ISSUE TITLE
ImportError: cannot import name 'NoOpContext'

### ISSUE THREAD

**Issue by dhirschfeld** [ipython/ipykernel#157]
I'm seeing the below error in ipykernel 4.3.1 with the latest 5.0 ipython from conda-forge when trying to run spyder.

AFAICS an ipykernel  compatible with the latest ipython hasn't been released?
```python-traceback
ERROR:tornado.application:Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x0000014E8A1A8D90>)
Traceback (most recent call last):
  File "C:\bin\Anaconda3\lib\site-packages\tornado\ioloop.py", line 600, in _run_callback
    ret = callback()
  File "C:\bin\Anaconda3\lib\site-packages\tornado\stack_context.py", line 275, in null_wrapper
    return fn(*args, **kwargs)
  File "C:\bin\Anaconda3\lib\site-packages\ipykernel\kernelbase.py", line 258, in enter_eventloop
    self.eventloop(self)
  File "C:\bin\Anaconda3\lib\site-packages\ipykernel\eventloops.py", line 73, in loop_qt4
    _notify_stream_qt(kernel, s)
  File "C:\bin\Anaconda3\lib\site-packages\ipykernel\eventloops.py", line 23, in _notify_stream_qt
    from IPython.core.interactiveshell import NoOpContext as context
ImportError: cannot import name 'NoOpContext'
```

**Comment by takluyver** [comment]
@Carreau was this unintentional? Do we want to make a quick release of ipykernel, or do an IPython 5.0.1 with this item back in place for imports?

**Comment by dhirschfeld** [comment]
From my POV it would seem to make sense that a major release of ipython would require a corresponding 5.0 version of ipykernel but if it's easier to release a 5.0.1 ipython that's fine by me. I guess there's the issue that the existing 5.0 ipython doesn't restrict it's dependencies to a compatible version of ipykernel ;/

**Comment by Carreau** [comment]
Nop, the NoOpContext got deprecated and come from IPython (`IPython/utils/contexts.py`), I took care of Deprecating them in IPython but probably missed a transitive import. 

I'll reintroduce in a `5.0.1`

**Comment by 3kwa** [comment]
When do you think you will be in a position to release that fix?

**Comment by takluyver** [comment]
This should also be fixed in ipykernel 4.4, which was just released. We'll still put it back in IPython in case there are other things depending on it.
